### PR TITLE
[IMP] pylint_odoo: Add new check docstring-first-line-empty

### PR DESCRIPTION
**Pull request by moylop260**
When odoo print a docstring just show the first line, but many times is empty

### CODE REVIEW

File: pylint_odoo/test_repo/broken_module/models/broken_model.py [hunk 1]
```diff
@@ -30,8 +30,11 @@ class TestModel(models.Model):
     )
 
     def my_method1(self, variable1):
-        #  Shouldn't show error of field-argument-translate
+        "Shouldn't show error of field-argument-translate"
         self.my_method2(_('hello world'))
 
     def my_method2(self, variable2):
+        '''
```
Comment (pedrobaeza) [inline pylint_odoo/test_repo/broken_module/models/broken_model.py]: Docstring should be with triple double-quote

Comment (moylop260) [inline pylint_odoo/test_repo/broken_module/models/broken_model.py]: Yes,
`broken_module` have allowed broke all lint's cases

Comment (pedrobaeza) [inline pylint_odoo/test_repo/broken_module/models/broken_model.py]: But in this case you are checking the beginning of the docstring with an empty line, not the correct quote. Expected errors should be then one more?

Comment (moylop260) [inline pylint_odoo/test_repo/broken_module/models/broken_model.py]: Yes,
We will can re-use this same method to create a new check `broken_module` has many lint issues for this matter.

**Comment by moylop260** [comment]
ping

**Comment by moylop260** [comment]
@nhomar 
Could you check this pr?

**Comment by guewen** [comment]
I'm not sure we want to start to code the pep257 checks here. At the very least we could use https://github.com/PyCQA/pydocstyle but I'm not sure we should check pep257. I think the usage of pydocstyle could be used if wanted by a PSC but should not be a requirement. I prefer to have docstrings that do not respect pep257 (as long as we don't use them for generating documentation) than no docstrings at all, and strict rules could discourage to write them at all.

**Comment by moylop260** [comment]
Thanks for comment.

This check don't force to use docstring or check all docstyles like as pydocstyle.
This pr is much more simple.

Odoo use the method string to show docstring in log.
But even show first line.
This is a custom way to show docstring just of odoo.

This check review that if you use a docstring, you should use the first line too.

Now, we can enable this check in the plugin (to use who decide use it), but we can disable it in MQT.

What do you think?

**Comment by guewen** [comment]
> Odoo use the method string to show docstring in log.
> But even show first line.
> This is a custom way to show docstring just of odoo.

When does it show the first line of the docstring in the logs?

**Comment by JulioSerna** [comment]
@guewen 

this log of runbot Odoo shows this log
runbot8 odoo com runbot static build 128945 8 0 8848af logs job_20_test_all txt

and that log is about this [test](https://github.com/odoo/odoo/blob/52484b0132ee3e0063dd67cc3c2eb434f99f5db8/openerp/addons/base/tests/test_menu.py#L10-L11)

as you can see in this case the log only shows the first line of docstring

**Comment by guewen** [comment]
> this log of runbot Odoo shows this log

This is not at all a custom behavior of Odoo, this is the behavior of Python's `unittest2`.

**Comment by moylop260** [comment]
Thats is right.
Then this pr help to avoid log empty jn this cases.

**Comment by moylop260** [comment]
@guewen 
If I add a validation to check just module/tests/* folder, Could you are agree?

**Comment by moylop260** [comment]
ping

**Comment by moylop260** [comment]
There is a python generic pep for this case [handling-docstring-indentation](https://www.python.org/dev/peps/pep-0257/#handling-docstring-indentation)

pep 0257 docstring conventions python org

And they give us a good code:
```python
# Since code is much more precise than words, here is an implementation of the algorithm:

def trim(docstring):
    if not docstring:
        return ''
    # Convert tabs to spaces (following the normal Python rules)
    # and split into a list of lines:
    lines = docstring.expandtabs().splitlines()
    # Determine minimum indentation (first line doesn't count):
    indent = sys.maxint
    for line in lines[1:]:
        stripped = line.lstrip()
        if stripped:
            indent = min(indent, len(line) - len(stripped))
    # Remove indentation (first line is special):
    trimmed = [lines[0].strip()]
    if indent < sys.maxint:
        for line in lines[1:]:
            trimmed.append(line[indent:].rstrip())
    # Strip off trailing and leading blank lines:
    while trimmed and not trimmed[-1]:
        trimmed.pop()
    while trimmed and not trimmed[0]:
        trimmed.pop(0)
    # Return a single string:
    return '\n'.join(trimmed)
```

Then I will close this one and we will work in follow issue: https://github.com/PyCQA/pylint/issues/868

**Pull request closed without merging**